### PR TITLE
[Merged by Bors] - feat: added topic deduplication mechanism 1/2

### PR DESCRIPTION
**Pull request by galibey**
This is the first part of [Dedup Impl PR](https://github.com/infinyon/fluvio/pull/3385) split for a more convenient review process.

### CODE REVIEW

File: crates/fluvio-protocol/src/link/error_code.rs [hunk 1]
```diff
@@ -182,6 +182,14 @@ pub enum ErrorCode {
     #[fluvio(tag = 9000)]
     #[error("a compression error occurred in the SPU")]
     CompressionError,
+
+    // Deduplication
+    #[fluvio(tag = 10000)]
```
Comment (sehz) [inline crates/fluvio-protocol/src/link/error_code.rs:187]: We should avoid adding new error codes until a new strategy is decided.  Not show stopper

File: k8-util/helm/fluvio-sys/templates/crd_topic.yaml [hunk 2]
```diff
@@ -90,6 +90,33 @@ spec:
                     maxPartitionSize:
                       type: integer
                       minimum: 2048
+                deduplication:
```
Comment (sehz) [inline k8-util/helm/fluvio-sys/templates/crd_topic.yaml:93]: Bit concerned about duplicating schema.   let's create issue to see if we can have better reusability

**Review by sehz — APPROVED** [review]
a couple of minor comments.  otherwise LGTM.  Nice work

File: crates/fluvio-controlplane-metadata/src/topic/deduplication.rs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    #[cfg_attr(
+        feature = "use_serde",
+        serde(
+            default,
+            skip_serializing_if = "Option::is_none",
+            with = "humantime_serde"
+        )
+    )]
+    pub age: Option<Duration>,
+}
```
Comment (digikata) [inline crates/fluvio-controlplane-metadata/src/topic/deduplication.rs:35]: Not something to change for this PR, but I wonder if there is some commonality with Lookback logic we can reuse between the two (slightly different) uses as a generic "record (pre-?)selection"

**Review by digikata — APPROVED** [review]
LGTM, nice work!

File: crates/fluvio-protocol/src/link/error_code.rs (hunk 1 above)
Comment (galibey) [reply crates/fluvio-protocol/src/link/error_code.rs:187]: Ok. Especially, on SPU/SC there is no much sense in enum of errors.

File: crates/fluvio-controlplane-metadata/src/topic/deduplication.rs (hunk 3 above)
Comment (galibey) [reply crates/fluvio-controlplane-metadata/src/topic/deduplication.rs:35]: Although Bounds and Lookback are common now, they might differ later. Bound covers both preselection(Lookback) and limits for runtime stream processing (via SmartModule init params).

**Pull request closed without merging**